src/qz/printer/action/WebApp.java

```diff
@@ -76,40 +72,43 @@ public void changed(ObservableValue<? extends Worker.State> ov, Worker.State old
                     base.getAttributes().setNamedItem(applied);
                 }
 
-                try {
-                    Reflect.on(webView).call("setZoom", pageZoom);
-                    log.trace("Zooming in by x{} for increased quality", pageZoom);
-                }
-                catch(ReflectException e) {
-                    log.warn("Unable zoom, using default quality");
-                    pageZoom = 1; //only zoom affects webView scaling
+                log.trace("Setting HTML page width to {}", pageWidth);
+                webView.setMinWidth(pageWidth);
+                webView.setPrefWidth(pageWidth);
+                webView.autosize();
+
+                //width was resized earlier (for responsive html), then calculate the best fit height
+                if (pageHeight <= 0) {
+                    String heightText = webView.getEngine().executeScript("Math.max(document.body.offsetHeight, document.body.scrollHeight)").toString();
+                    pageHeight = Double.parseDouble(heightText);
                 }
 
-                log.trace("Setting HTML page width to {}", (pageWidth * pageZoom));
-                webView.setMinWidth(pageWidth * pageZoom);
-                webView.setPrefWidth(pageWidth * pageZoom);
+                log.trace("Setting HTML page height to {}", pageHeight);
+                webView.setMinHeight(pageHeight);
+                webView.setPrefHeight(pageHeight);
                 webView.autosize();
 
-                //we have to resize the width first, for responsive html, then calculate the best fit height
-                final PauseTransition resize = new PauseTransition(Duration.millis(100));
-                resize.setOnFinished(new EventHandler<ActionEvent>() {
+                //without this runlater, the first capture is missed and all following captures are offset
+                Platform.runLater(new Runnable() {
                     @Override
-                    public void handle(ActionEvent actionEvent) {
-                        if (pageHeight <= 0) {
-                            String heightText = webView.getEngine().executeScript("Math.max(document.body.offsetHeight, document.body.scrollHeight)").toString();
-                            pageHeight = Double.parseDouble(heightText);
-                        }
-
-                        log.trace("Setting HTML page height to {}", (pageHeight * pageZoom));
-                        webView.setMinHeight(pageHeight * pageZoom);
-                        webView.setPrefHeight(pageHeight * pageZoom);
-                        webView.autosize();
-
-                        snap.playFromStart();
+                    public void run() {
+                        log.debug("Attempting image capture");
+
+                        SnapshotParameters sparams = new SnapshotParameters();
+                        sparams.setTransform(new Scale(pageZoom, pageZoom));
```

Comment:
On Java 8 and Java 11 using 4x zoom, it appears to just make the picture bigger without improving quality... :(

![image](https://user-images.githubusercontent.com/6345473/74216592-e62e5400-4c72-11ea-9342-e1da89bdeb60.png)